control_panel_api/management/commands/load_kube_users.py

```diff
@@ -0,0 +1,42 @@
+import base64
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db.utils import IntegrityError
+from kubernetes import config, client
+
+from control_panel_api.models import User
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Import Kubernetes user-secrets data into the User.  Uses KUBECONFIG envvar or defaults to ~/.kube/config"
+
+    USER_SECRETS_NAME = 'user-secrets'
+
+    def add_arguments(self, parser):
+        pass
+
+    def handle(self, *args, **options):
+        config.load_kube_config()
+        v1 = client.CoreV1Api()
+        results = v1.list_secret_for_all_namespaces()
+
+        user_secrets = [item.data for item in results.items if item.metadata.name == Command.USER_SECRETS_NAME]
```

Comment:
Couldn't find the documentation for this module but I was wondering if there is a way to get the list of secrets matching this directly from k8s.

Far from a k8s expert but I guess `list_secret_for_all_namespaces()` will call [this k8s endpoint](https://kubernetes.io/docs/api-reference/v1.6/#list-all-namespaces-211).

Nevermind, [doesn't look like fieldSelector is supported very well](https://github.com/kubernetes/kubernetes/issues/15128) a part from pods at the moment.